src/main/java/com/hubspot/jinjava/util/EagerReconstructionUtils.java

```diff
@@ -104,7 +104,9 @@ public static EagerExecutionResult executeInChildContext(
           .filter(e -> !metaContextVariables.contains(e.getKey()))
           .filter(
             entry ->
-              !(entry.getValue() instanceof DeferredValue) && entry.getValue() != null
+              !(entry.getValue() instanceof DeferredValue) &&
+              entry.getValue() != null &&
+              getObjectOrHashCode(entry.getValue()) != null
```

Comment:
`getObjectOrHasCode()` is sometimes an expensive call, so it would be good to only have to call it once:
```
          .filter(e -> !(e.getValue() instanceof DeferredValue))
          .map(
            e ->
              new AbstractMap.SimpleImmutableEntry<>(
                e.getKey(),
                getObjectOrHashCode(e.getValue())
              )
          )
          .filter(e -> e.getValue() != null)
          .collect(
            Collectors.toMap(Entry::getKey, Entry::getValue)
          )

Reply:
Thanks for pointing that out. I have changed that. Could you please have another look?